elide-datastore/elide-datastore-jpa/src/main/java/com/yahoo/elide/datastores/jpa/transaction/AbstractJpaTransaction.java

```diff
@@ -295,4 +321,45 @@ public Object getRelation(
     public void cancel(RequestScope scope) {
         jpaTransactionCancel.accept(em);
     }
+
+
+    /**
+     * Creates an entity graph for the first encountered to Many relationship that meets the following criteria
+     *  * The relationship has to be present in Entity Projection
+     *  * The relationship type should be ToMany and not computed relationship
+     *  * The relationship Entity should not be presnet in dirtyResource.
+     *      (no mutable operation is done on the relationship entity in this request)
+     *  * If the relationship Entity is presnet in dirtyResource,
+     *      then the relationshipship should have CASCADE.ALL cascade type.
+     * @param scope
+     * @param projection
+     * @return
+     */
+    private EntityGraph createEntityGraph(RequestScope scope, EntityProjection projection) {
+        EntityDictionary dictionary = scope.getDictionary();
+        Class<?> entityClass = projection.getType();
+        List<String> relationshipNames = dictionary.getRelationships(entityClass);
+        for (String relationshipName : relationshipNames) {
+            if (!projection.getIncludedRelationsName().contains(relationshipName)) {
+                continue;
+            }
+            RelationshipType type = dictionary.getRelationshipType(entityClass, relationshipName);
+
+            if (type.isToMany() && !type.isComputed()) {
+                // To Many Relationship will always be collection type.
+                String relationshipClassName = dictionary.getJsonAliasFor(
+                        dictionary.getParameterizedType(entityClass, relationshipName));
+                boolean isDirtyResource = StreamSupport.stream(scope.getDirtyResources().spliterator(), false)
+                        .anyMatch(resource -> resource.getType().equals(relationshipClassName));
+
+                if (!isDirtyResource || dictionary.cascadeAll(entityClass, relationshipName)) {
+                    EntityGraph entityGraph = em.createEntityGraph(entityClass);
+                    entityGraph.addAttributeNodes(relationshipName);
```

Comment:
Why not add all the projection attributes and relationships?  Also, can this recurse?